dm/dm.go

```diff
@@ -72,13 +73,15 @@ var usage = func() {
 	fmt.Fprintln(os.Stderr, "Flags:")
 	flag.PrintDefaults()
 	fmt.Fprintln(os.Stderr)
-	os.Exit(1)
+	fmt.Fprintln(os.Stderr, "--stdin requires a file name and either the file contents or a tar archive containing the named file.")
+	fmt.Fprintln(os.Stderr, "        a tar archive may include any additional files referenced directly or indirectly by the named file.")
+	panic("\n")
 }
 
 func getGitRegistry() *registry.GithubRegistry {
 	s := strings.Split(*template_registry, "/")
 	if len(s) < 2 {
-		log.Fatalf("invalid template registry: %s", *template_registry)
+		panic(fmt.Errorf("invalid template registry: %s", *template_registry))
```

Comment:
idiomatic for what? I have read that and panic/recover is for cleanup. The only thing we do is call log fatal in the recover? I guess I'm not explaining my question properly but I'm confused why we set up added complexity for simply calling log.fatal in the recover method when we could just leave the calls as log.fatal as they are and nothing would change. As in, what added benefit do we get by introducing more code?